compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IterableAnalyzer.java

```diff
@@ -414,4 +414,53 @@ BType calculateType(Operation operation, BType type) {
             return symTable.errType;
         }
     }
+
+    public void validateIterableContext(IterableContext context) {
+        final Operation lastOperation = context.operations.getLast();
+        final List<BType> expectedTypes = lastOperation.expectedTypes;
+        final List<BType> resultTypes = lastOperation.resultTypes;
+        if (expectedTypes.size() == 0 && resultTypes.size() == 0) {
+            context.resultType = symTable.noType;
+            return;
+        }
+        if (expectedTypes.size() == 0) {
```

Comment:
Instead of size can we use isEmpty here? Also instead of checking size at different places,can we assign isEmpty result to a variable at the top and use it for all the if conditions?